oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -765,9 +834,27 @@ int mca_spml_ucx_fence(shmem_ctx_t ctx)
 
 int mca_spml_ucx_quiet(shmem_ctx_t ctx)
 {
+    int flush_get_data;
     int ret;
+    unsigned i;
+    int idx;
     mca_spml_ucx_ctx_t *ucx_ctx = (mca_spml_ucx_ctx_t *)ctx;
 
+    if (mca_spml_ucx.synchronized_quiet) {
+        for (i = 0; i < ucx_ctx->put_proc_count; i++) {
+            idx = ucx_ctx->put_proc_indecies[i];
+            ret = mca_spml_ucx_get_nb(ctx,
+                                      ucx_ctx->ucp_peers[idx].mkeys->super.super.va_base,
+                                      sizeof(flush_get_data), &flush_get_data, idx, NULL);
+            if (OMPI_SUCCESS != ret) {
+                oshmem_shmem_abort(-1);
+                return ret;
+            }
+        }
+        opal_bitmap_clear_all_bits(&ucx_ctx->put_op_bitmap);
```

Comment:
need to clear bits in prev loop. otherwise it's O(n) operation here

Reply:
added to atomics too